### PR TITLE
Add gatheringData prop to ReportTable component.

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue:

- #4908 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by tofumatt — CHANGES_REQUESTED** [review]
I set Search Console to be in a Gathering Data state but it still is showing data, which I think is unrelated but just wanted to flag it 😅 

CleanShot 2022-03-18 at 01 39 49

----

Looks good though! I think some VRTs are failing here—could you update them? Then this should be good-to-go.

**Comment by asvinb** [comment]
@tofumatt Can you review the changes again, especially these lines:
https://github.com/google/site-kit-wp/pull/4952/files#diff-936b7ee65e0faeaa4e88d0a40efb31a526270e87a94b3ec63d28a8abad479f79L106-L108
I'am not sure why we had it in first place 🤔

File: assets/js/components/wp-dashboard/WPDashboardPopularPages.js [hunk 1]
```diff
@@ -103,15 +106,11 @@ export default function WPDashboardPopularPages( props ) {
 		return <PreviewTable rows={ 6 } />;
 	}
 
-	if ( ! isGatheringData ) {
-		return null;
-	}
```
Comment (tofumatt) [inline assets/js/components/wp-dashboard/WPDashboardPopularPages.js:108]: Hmm, odd, yeah, not sure why this is here… 🤔 

Looks like you added it 😆  https://github.com/google/site-kit-wp/commit/53a2f75f4813bf285415f2d35c77a5515db0ed9d#diff-936b7ee65e0faeaa4e88d0a40efb31a526270e87a94b3ec63d28a8abad479f79R102-R104

Anyway, I think it's best to remove 👍🏻

**Review by tofumatt — APPROVED** [review]
Much nicer! 👍🏻 

CleanShot 2022-03-21 at 18 30 54

**Pull request merged**